torch/csrc/jit/codegen/cuda/kernel_ir_printer.h

```diff
@@ -91,10 +95,26 @@ class TORCH_CUDA_API IrPrinter : private kir::IrVisitor {
 
   // The set of values used by the current top-level IR node
   std::unordered_set<const kir::Val*> uses_;
+
+  // If the definition of all inputs to an expression haven't been printed
+  // already implicit_definition_ = true will print them before printing the
+  // requested node.
+  bool implicit_definition_ = true;
 };
 
-//! Returns the string representation of a Kernel IR node
-std::string toString(const kir::Node* stmt);
+//! Returns the string representation of a Kernel IR node. If the definition of
+//! all inputs to an expression haven't been printed already
+//! implicit_definition_ = true will print them before printing the requested
+//! node.
+std::string toString(const kir::Node* stmt, bool implicit_definitions = true);
+
+//! Returns the string representation of a vector of kir::Expr, convenient
+//! debugm echanism during lowering. If the definition of all inputs to an
+//! expression haven't been printed already implicit_definition_ = true will
+//! print them before printing the requested node.
+std::string toString(
```

Comment:
we should probably have a generic toString(vector<T>), but for now this would do